docs/root/intro/arch_overview/http/upgrades.rst

```diff
@@ -94,17 +94,31 @@ will synthesize 200 response headers, and then forward the TCP data as the HTTP
 For an example of proxying connect, please see :repo:`configs/proxy_connect.yaml <configs/proxy_connect.yaml>`
 For an example of terminating connect, please see :repo:`configs/terminate_connect.yaml <configs/terminate_connect.yaml>`
 
-Tunneling TCP over HTTP/2
-^^^^^^^^^^^^^^^^^^^^^^^^^
-Envoy also has support for transforming raw TCP into HTTP/2 CONNECT requests. This can be used to
-proxy multiplexed TCP over pre-warmed secure connections and amortize the cost of any TLS handshake.
-An example set up proxying SMTP would look something like this
+Tunneling TCP over HTTP
+^^^^^^^^^^^^^^^^^^^^^^^
+Envoy also has support for tunneling raw TCP over HTTP CONNECT requests. Find
+below some usage scenarios.
+
+HTTP/2 CONNECT can be used to proxy multiplexed TCP over pre-warmed secure connections and amortize the cost of any TLS
+handshake.
+An example set up proxying SMTP would look something like this:
 
 [SMTP Upstream] --- raw SMTP --- [L2 Envoy]  --- SMTP tunneled over HTTP/2  --- [L1 Envoy]  --- raw SMTP  --- [Client]
 
 Examples of such a set up can be found in the Envoy example config :repo:`directory <configs/>`
-If you run `bazel-bin/source/exe/envoy-static --config-path configs/encapsulate_in_connect.yaml --base-id 1`
+If you run `bazel-bin/source/exe/envoy-static --config-path configs/encapsulate_in_http2_connect.yaml --base-id 1`
 and `bazel-bin/source/exe/envoy-static --config-path  configs/terminate_connect.yaml`
 you will be running two Envoys, the first listening for TCP traffic on port 10000 and encapsulating it in an HTTP/2
 CONNECT request, and the second listening for HTTP/2 on 10001, stripping the CONNECT headers, and forwarding the
 original TCP upstream, in this case to google.com.
+
+HTTP/1.1 CONNECT can be used to have TCP client connecting to its own
+destination passing through an HTTP proxy server (e.g. corporate proxy):
+
+[HTTP Server] --- raw HTTP --- [Upstream HTTP Proxy]  --- HTTP tunneled over HTTP/1.1  --- [Envoy]  --- raw HTTP  --- [HTTP Client]
+
+Examples of such a set up can be found in the Envoy example config :repo:`directory <configs/>`
+If you run `bazel-bin/source/exe/envoy-static --config-path configs/encapsulate_in_http1_connect.yaml --base-id 1`
+you will be running Envoy listening for TCP traffic on port 10000 and encapsulating it in an HTTP/1.1
+CONNECT addressed to an HTTP proxy running on localhost and listenig on port
+10001, having as a final destination host.com on port 443.
```

Comment:
As a first time reader this seems a bit confusing: which proxy are we configuring with the provided config? Both? Just the one labeled 'Envoy'?

Reply:
In this scenario, I made the assumption that the `Upstream HTTP Proxy` is not an Envoy instance, that's why I only labeled the proxy that is tunneling the traffic with `Envoy` and just provided one configuration `configs/encapsulate_in_http1_connect.yaml`.
The main driver of this PR is to support use cases where the upstream HTTP proxy is supporting HTTP/1 only e.g. #11308
Do you think I should make this more clear?

Reply:
Having re-read this a few times i realize my confusion stems from the fact that the flow is server -> client, not client -> server, and I was originally thinking of the encapsulation flow as originating from the client. 

I think I would still have this example make use of two Envoy instances (assuming we support this) and then mention that this can be used when the upstream is another proxy that does not support HTTP/2 (to justify why you would ever use HTTP/1.1). I think this would be clearer and would make the example complete, as its not relying on the reader knowing how to set up another local proxy.

Reply:
yep, this could make sense. If it's not a problem for you I could address this with a follow-up PR as I still have some work to do on this feature.

Reply:
I did this already as I had to do another change.